datafusion/functions/src/datetime/date_part.rs

```diff
@@ -157,28 +114,28 @@ impl ScalarUDFImpl for DatePartFunc {
             ColumnarValue::Scalar(scalar) => scalar.to_array()?,
         };
 
-        let arr = match date_part.to_lowercase().as_str() {
-            "year" => extract_date_part!(&array, temporal::year),
-            "quarter" => extract_date_part!(&array, temporal::quarter),
-            "month" => extract_date_part!(&array, temporal::month),
-            "week" => extract_date_part!(&array, temporal::week),
-            "day" => extract_date_part!(&array, temporal::day),
-            "doy" => extract_date_part!(&array, temporal::doy),
-            "dow" => extract_date_part!(&array, temporal::num_days_from_sunday),
-            "hour" => extract_date_part!(&array, temporal::hour),
-            "minute" => extract_date_part!(&array, temporal::minute),
-            "second" => extract_date_part!(&array, seconds),
-            "millisecond" => extract_date_part!(&array, millis),
-            "microsecond" => extract_date_part!(&array, micros),
-            "nanosecond" => extract_date_part!(&array, nanos),
-            "epoch" => extract_date_part!(&array, epoch),
-            _ => exec_err!("Date part '{date_part}' not supported"),
-        }?;
+        let arr = match part.to_lowercase().as_str() {
```

Comment:
https://github.com/apache/arrow-rs/pull/5318

Reply:
As above, I don't understand the reference to the JSON writer PR

The changes in this module look more like switching to use the `date_part` kernels that @Jefffrey  added in https://github.com/apache/arrow-rs/pull/5319 and a cleanup of the code to use the `unary` kernel more effectively (the changes look good to me)

Reply:
Yes this was a copypasta